src/server.c

```diff
@@ -1085,7 +1085,10 @@ void clientsCron(void) {
 
     /* Pause the IO threads that are processing clients, to let us access clients
      * safely. In order to avoid increasing CPU usage by pausing all threads when
-     * there are too many io threads, we pause io threads in multiple batches. */
+     * there are too many io threads, we pause io threads in multiple batches.
+     * NOTE: If the number of IO threads exceeds 8 (CLIENTS_CRON_PAUSE_IOTHREAD),
+     * processing all clients may take a few seconds, potentially breaking the
+     * assumption that all clients are processed within 1 second. */
```

Comment:
is it few seconds? do we have guarantee that we'll eventually process all of them? or could it be that a certain client will be skipped again and again?

Reply:
we do clients cron every 100ms, we alternately process clients owned by different batches of threads.
we do not have any guarantee, in some edge cases, it may take a lot of time, or it may never be processed